nservicebus/hosting/critical-errors.md

```diff
@@ -46,10 +46,16 @@ snippet:CustomHostErrorHandlingAction
 
 ## When to override the default action
 
-The default action should be overridden whenever that default does not meet the specific hosting requirements. For example
+When self hosting the should be overriden in order to exit the current process unless you are using the [NServiceBus Host](/nservicebus/hosting/nservicebus-host) which already has a implementation that exits the process.
 
- * If using [NServiceBus Host](/nservicebus/hosting/nservicebus-host), and wish to take a custom action before the endpoint process is killed.
- * If self hosting the process can be shut down the process via [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) and re-start the process once the root cause has been diagnosed.
+Consider the following things when implementing a critical error callback:
+
+- Wrap any tasks in a `try..finally`, making sure the process will exit if any tasks fail.
+- Flush any loggers used which makes sure any unwritten tail of the log is persisted as [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) will immediately exit the process
+- Flush any state/caches used if you normally would flush such data structure in a certain interval.
```

Comment:
No, not cleared, that in memory state like a cache is persisted. It isn't applicable. If a customer is doing such a thing at normal application shutdown then they should do it here too.